src/pybind/mgr/dashboard/frontend/src/app/shared/components/usage-bar/usage-bar.component.ts

```diff
@@ -1,53 +1,115 @@
-      this.usedPercentage = this.total > 0 ? (this.used / this.total) * 100 : 0;
-      this.freePercentage = 100 - this.usedPercentage;
-    } else {
-      if (this.used) {
-        this.used = this.used.slice(0, -1);
-        this.usedPercentage = Number(this.used);
-        this.freePercentage = 100 - this.usedPercentage;
-      } else {
-        this.usedPercentage = 0;
+export class UsageBarComponent implements OnInit {
+  @Input() total: number = 0;
+  @Input() used: number = 0;
+  @Input() title: string = $localize`Usage`;
+  @Input() warningThreshold: number = 0.9;
+  @Input() errorThreshold: number = 0.95;
+  @Input() isBinary: boolean = true;
+  @Input() decimal: number = 2;
+  @Input() proportional: boolean = true;                                                                                                         ;
+
+  data: ChartTabularData = [];
+  options: MeterChartOptions = {
+    legend: {
+      enabled: false                                                                                      
+    },
+    meter: {
+      status: {
+        ranges: []
       }
-    }
+    },
+    tooltip: {
+      enabled: true
+    },
+    toolbar: {
+      enabled: false
+    },
+    height: '2rem',
+    width: '15rem'
+  };
+
+  constructor(
+    private elementRef: ElementRef,
+    private cssHelper: CssHelper,
+    private dimlessPipe: DimlessPipe,
+    private dimlessBinaryPipe: DimlessBinaryPipe
+  ) {}
+
+  ngOnInit(): void {
+    const usedPercentage = this.calculateUsed(this.used, this.total) * 100;
+    this.data = [
+      {
+        group: this.title,
+        value: usedPercentage
+      }
+    ];
+    const thresholds: { range: [number, number]; status: Statuses | string }[] = [
+      {
+        range: [0, this.warningThreshold * 100],
+        status: Statuses.SUCCESS
+      },
+      {
+        range: [this.warningThreshold * 100, this.errorThreshold * 100],
+        status: Statuses.WARNING
+      },
+      {
+        range: [this.errorThreshold * 100, 100],
+        status: Statuses.DANGER
+      }
+    ];
+
+    const meterBase = {
+      status: { ranges: thresholds },
+      ...(this.proportional && {
+        proportional: {
+          total: this.total,
+          totalFormatter: () =>
+            this.isBinary
+              ? $localize`${this.dimlessBinaryPipe.transform(this.total, this.decimal)} total`
+              : $localize`${this.dimlessPipe.transform(this.total, this.decimal)} total`,
+          breakdownFormatter: () => {
+            const used = this.isBinary
+              ? this.dimlessBinaryPipe.transform(this.used, this.decimal)
+              : this.dimlessPipe.transform(this.used, this.decimal);
+            const available = this.isBinary
+              ? this.dimlessBinaryPipe.transform(this.total - this.used, this.decimal)
+              : this.dimlessPipe.transform(this.total - this.used, this.decimal);
+            return $localize`${used} used (${available} available)`;
```

Comment:
```suggestion
            return $localize`${used} used`;
```

Either we show used/total or used/available. Too much text is getting trimmed now.

Reply:
needs more discussion IMO